Formula/llvm.rb

```diff
@@ -275,10 +279,59 @@ def caveats
     # related to https://github.com/Homebrew/legacy-homebrew/issues/47149
     system "#{bin}/clang++", "-v",
            "-std=c++11", "-stdlib=libc++", "test.cpp", "-o", "testlibc++",
-           "-Wl,-rpath,#{opt_lib}"
+           "-L#{opt_lib}", "-Wl,-rpath,#{opt_lib}"
     on_macos { assert_includes MachO::Tools.dylibs("testlibc++"), "#{opt_lib}/libc++.1.dylib" }
     assert_equal "Hello World!", shell_output("./testlibc++").chomp
 
+    on_linux do
+      # Link installed libc++, libc++abi, and libunwind archives both into
+      # a position independent executable (PIE), as well as into a fully
+      # position independent (PIC) DSO for things like plugins that export
+      # a C-only API but internally use C++.
+      #
+      # FIXME: It'd be nice to be able to use flags like `-static-libstdc++`
+      # together with `-stdlib=libc++` (the latter one we need anyways for
+      # headers) to achieve this but those flags don't set up the correct
+      # search paths or handle all of the libraries needed by `libc++` when
+      # linking statically.
+
+      system "#{bin}/clang++", "-v", "-o", "test_pie_runtimes",
+             "-pie", "-fPIC", "test.cpp", "-L#{opt_lib}",
+             "-stdlib=libc++", "-rtlib=compiler-rt", "-unwindlib=libunwind",
+             "-static-libstdc++", "-static-libgcc", "-lpthread", "-ldl"
+      assert_equal "Hello World!", shell_output("./test_pie_runtimes").chomp
+      Pathname.new("test_pie_runtimes").dynamically_linked_libraries.each do |lib|
```

Comment:
If you're applying my suggestion above, let's make this consistent:
```suggestion
      (testpath/"test_pie_runtimes").dynamically_linked_libraries.each do |lib|
```

Reply:
I'm a total Ruby novice, so any suggestions on better code are very much welcome. =D